src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -127,7 +129,13 @@ def acr_task_create(cmd,  # pylint: disable=too-many-locals
 
     Task, PlatformProperties, AgentProperties, TriggerProperties = cmd.get_models(
         'Task', 'PlatformProperties', 'AgentProperties', 'TriggerProperties')
+
+    identity = None
+    if assign_identity is not None:
```

Comment:
nit: move this check inside `_build_identities_info` so you don't have to initialize `identity` to None and have this check, and just have 
```
    task_create_parameters = Task(
        identity=_build_identities_info(cmd, assign_identity),
...
```